### PR TITLE
componentization of running over atoms (ie crazy mulebots)

### PR DESCRIPTION
**Pull request by ShizCalev**
Framework so it can be added to vehicles later.

Requires #43250

:cl: ShizCalev
fix: Fixed logging of mulebots running over mobs not reporting the mob's new health correctly.
fix: Running over a mob with a mulebot will now correctly spawn blood on the ground.
tweak: Mobs will now scream when run over.
refactor: Refactored the running over of mobs into a component that supports running over other things so it can easily be added to things other than mulebots. 
fix: Mulebots will no longer get bloody if they run over a mob with no blood.
/:cl:

### CODE REVIEW

File: code/datums/components/run_over_atom.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/atom/movable/proc/after_running_over(atom/movable/AM)
+	return
+
+/mob/living/simple_animal/bot/mulebot/after_running_over(atom/movable/AM)
+	bloodiness += 4
+
+/atom/movable/proc/get_run_over(atom/movable/AM, _damage_amount, run_over_message)
+	return
+
+/mob/living/carbon/get_run_over(atom/movable/AM, _damage_amount, run_over_message)
```
Comment (optimumtact) [inline code/datums/components/run_over_atom.dm]: shouldn't this live on the datum, it seems like you're almost getting seperation of concerns but then falling down at the last steps

Comment (ShizCalev) [reply code/datums/components/run_over_atom.dm]: Have an example of what you mean by chance? Doing this to figure out how components works, really.

Comment (optimumtact) [reply code/datums/components/run_over_atom.dm]: as in the component should deliver the damage to the object this proc shouldn't exist at all, or be a proc on the datum that just takes the target mob.

**Comment by ShizCalev** [comment]
Will complete once I've moved.

**Pull request closed without merging**